Compilation failure when compiling icub-main devel branch against yarp latest master branch (2021/09/02)

### Bug description

Compiling latest icub-main devel against YARP master branch fails with error:

~~~
2021-09-02T02:41:33.3567443Z [334/458] Building CXX object src/simulators/iCubSimulation/CMakeFiles/iCub_SIM.dir/odesdl/MS3D.cpp.o
2021-09-02T02:41:33.3568333Z FAILED: src/simulators/iCubSimulation/CMakeFiles/iCub_SIM.dir/odesdl/MS3D.cpp.o 
2021-09-02T02:41:33.3575273Z /usr/bin/c++ -DICUB_SIM_ENABLE_FAKE -DICUB_SIM_ENABLE_ODESDL -DOMIT_LOGPOLAR -DUSE_GLUT -D_USE_MATH_DEFINES -DdDOUBLE -I/home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation -I/home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/wrapper -I/home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/fake -I/usr/include/SDL -I/home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl -I/home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/libraries/iKin/include -I/home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/libraries/ctrlLib/include -I/home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/libraries/skinDynLib/include -isystem /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/include -O3 -DNDEBUG -fPIE -std=c++17 -MD -MT src/simulators/iCubSimulation/CMakeFiles/iCub_SIM.dir/odesdl/MS3D.cpp.o -MF src/simulators/iCubSimulation/CMakeFiles/iCub_SIM.dir/odesdl/MS3D.cpp.o.d -o src/simulators/iCubSimulation/CMakeFiles/iCub_SIM.dir/odesdl/MS3D.cpp.o -c /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.cpp
2021-09-02T02:41:33.3581518Z In file included from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.h:40,
2021-09-02T02:41:33.3583084Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.cpp:28:
2021-09-02T02:41:33.3584113Z /usr/include/SDL/SDL_opengl.h:116: warning: "GL_GLEXT_VERSION" redefined
2021-09-02T02:41:33.3584686Z   116 | #define GL_GLEXT_VERSION 29
2021-09-02T02:41:33.3585030Z       | 
2021-09-02T02:41:33.3585432Z In file included from /usr/include/GL/gl.h:2050,
2021-09-02T02:41:33.3586034Z                  from /usr/include/SDL/SDL_opengl.h:46,
2021-09-02T02:41:33.3587168Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.h:40,
2021-09-02T02:41:33.3588699Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.cpp:28:
2021-09-02T02:41:33.3589741Z /usr/include/GL/glext.h:54: note: this is the location of the previous definition
2021-09-02T02:41:33.3590335Z    54 | #define GL_GLEXT_VERSION 20190805
2021-09-02T02:41:33.3590703Z       | 
2021-09-02T02:41:33.3591747Z In file included from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.cpp:28:
2021-09-02T02:41:33.3593445Z /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.h:84:5: error: reference to ‘byte’ is ambiguous
2021-09-02T02:41:33.3594377Z    84 |     byte m_flags;
2021-09-02T02:41:33.3594692Z       |     ^~~~
2021-09-02T02:41:33.3595691Z In file included from /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/include/yarp/conf/numeric.h:19,
2021-09-02T02:41:33.3597208Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/include/yarp/os/ConnectionReader.h:10,
2021-09-02T02:41:33.3598828Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/include/yarp/os/PortReaderCreator.h:10,
2021-09-02T02:41:33.3600306Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/include/yarp/os/Port.h:13,
2021-09-02T02:41:33.3601731Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/include/yarp/os/BufferedPort.h:11,
2021-09-02T02:41:33.3603276Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/VideoTexture.h:34,
2021-09-02T02:41:33.3604882Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/rendering.h:33,
2021-09-02T02:41:33.3606435Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.h:52,
2021-09-02T02:41:33.3607951Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.cpp:28:
2021-09-02T02:41:33.3609312Z /usr/include/c++/9/cstddef:69:14: note: candidates are: ‘enum class std::byte’
2021-09-02T02:41:33.3609977Z    69 |   enum class byte : unsigned char {};
2021-09-02T02:41:33.3610355Z       |              ^~~~
2021-09-02T02:41:33.3611444Z In file included from /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.cpp:28:
2021-09-02T02:41:33.3613140Z /home/runner/work/robotology-superbuild/robotology-superbuild/src/ICUB/src/simulators/iCubSimulation/odesdl/MS3D.h:73:23: note:                 ‘typedef unsigned char byte’
2021-09-02T02:41:33.3614131Z    73 | typedef unsigned char byte;
2021-09-02T02:41:33.3614523Z       |                       ^~~~
~~~


### Steps to reproduce

The easiest way is to just compile the robotology-superbuild with `-DROBOTOLOGY_PROJECT_TAGS:STRING=Unstable` , see https://github.com/robotology/robotology-superbuild/issues/872 .

### Expected behavior

I would expect the compilation to be successful.

### Example repository

_No response_

### Additional context

_No response_

--- a/src/simulators/iCubSimulation/odesdl/MS3D.cpp
+++ b/src/simulators/iCubSimulation/odesdl/MS3D.cpp
@@ -116,10 +116,10 @@ bool Model::loadModelData( const char *filename )
     if ( pHeader->m_version < 3 )
         return false; // "Unhandled file version. Only Milkshape3D Version 1.3 and 1.4 is supported." );
 
-    int nVertices = *( word* )pPtr; 
+    int nVertices = *( ms3d::word* )pPtr; 
     m_numVertices = nVertices;
     m_pVertices = new Vertex[nVertices];
-    pPtr += sizeof( word );
+    pPtr += sizeof( ms3d::word );
 
     int i;
     for ( i = 0; i < nVertices; i++ )
@@ -130,10 +130,10 @@ bool Model::loadModelData( const char *filename )
         pPtr += sizeof( MS3DVertex );
     }
 
-    int nTriangles = *( word* )pPtr;
+    int nTriangles = *( ms3d::word* )pPtr;
     m_numTriangles = nTriangles;
     m_pTriangles = new Triangle[nTriangles];
-    pPtr += sizeof( word );
+    pPtr += sizeof( ms3d::word );
 
     for ( i = 0; i < nTriangles; i++ )
     {
@@ -149,22 +149,22 @@ bool Model::loadModelData( const char *filename )
 
 
 
-    int nGroups = *( word* )pPtr;
+    int nGroups = *( ms3d::word* )pPtr;
     m_numMeshes = nGroups;
     m_pMeshes = new Mesh[nGroups];
-    pPtr += sizeof( word );
+    pPtr += sizeof( ms3d::word );
     for ( i = 0; i < nGroups; i++ )
     {
-        pPtr += sizeof( byte );	// flags
+        pPtr += sizeof( ms3d::byte );	// flags
         pPtr += 32;				// name
 
-        word nTriangles = *( word* )pPtr;
-        pPtr += sizeof( word );
+        ms3d::word nTriangles = *( ms3d::word* )pPtr;
+        pPtr += sizeof( ms3d::word );
         int *pTriangleIndices = new int[nTriangles];
         for ( int j = 0; j < nTriangles; j++ )
         {
-            pTriangleIndices[j] = *( word* )pPtr;
-            pPtr += sizeof( word );
+            pTriangleIndices[j] = *( ms3d::word* )pPtr;
+            pPtr += sizeof( ms3d::word );
         }
 
         char materialIndex = *( char* )pPtr;
@@ -175,10 +175,10 @@ bool Model::loadModelData( const char *filename )
         m_pMeshes[i].m_pTriangleIndices = pTriangleIndices;
     }
 
-    int nMaterials = *( word* )pPtr;
+    int nMaterials = *( ms3d::word* )pPtr;
     m_numMaterials = nMaterials;
     m_pMaterials = new Material[nMaterials];
-    pPtr += sizeof( word );
+    pPtr += sizeof( ms3d::word );
     for ( i = 0; i < nMaterials; i++ )
     {
         MS3DMaterial *pMaterial = ( MS3DMaterial* )pPtr;

--- a/src/simulators/iCubSimulation/odesdl/MS3D.h
+++ b/src/simulators/iCubSimulation/odesdl/MS3D.h
@@ -70,8 +70,10 @@
 #	error you must byte-align these structures with the appropriate compiler directives
 #endif
 
-typedef unsigned char byte;
-typedef unsigned short word;
+namespace ms3d {
+  typedef unsigned char byte;
+  typedef unsigned short word;
+}
 
 struct MS3DHeader               // File Header
 {
@@ -81,20 +83,20 @@ struct MS3DHeader               // File Header
 
 struct MS3DVertex               // Vertex info
 {
-    byte m_flags;
+    ms3d::byte m_flags;
     float m_vertex[3];
     char m_boneID;
-    byte m_refCount;
+    ms3d::byte m_refCount;
 } PACK_STRUCT;
 
 struct MS3DTriangle             // Triangle info
 {
-    word m_flags;
-    word m_vertexIndices[3];
+    ms3d::word m_flags;
+    ms3d::word m_vertexIndices[3];
     float m_vertexNormals[3][3];
     float m_s[3], m_t[3];
-    byte m_smoothingGroup;
-    byte m_groupIndex;
+    ms3d::byte m_smoothingGroup;
+    ms3d::byte m_groupIndex;
 } PACK_STRUCT;
 
 struct MS3DMaterial             // Material info
@@ -108,7 +110,7 @@ struct MS3DMaterial             // Material info
     float m_emissive[4];
     float m_shininess;          // 0.0f - 128.0f
     float m_transparency;       // 0.0f - 1.0f
-    byte m_mode;                // 0, 1, 2 (unused now)
+    ms3d::byte m_mode;                // 0, 1, 2 (unused now)
     char m_texture[128];
     char m_alphamap[128];
 } PACK_STRUCT;

--- a/src/tools/skinManagerGui/CMakeLists.txt
+++ b/src/tools/skinManagerGui/CMakeLists.txt
@@ -36,6 +36,9 @@ add_executable(skinManagerGui WIN32    ${skinManagerGui_SRCS}
                                        ${skinManagerGui_HDRS}
                                        ${skinManagerGui_QRC_GEN_SRCS}
                                        ${skinManagerGui_UI_GEN_SRCS})
+if(WIN32)
+  target_compile_definitions(skinManagerGui PUBLIC _HAS_STD_BYTE=0)
+endif()
 target_link_libraries(skinManagerGui    YARP::YARP_OS
                                         YARP::YARP_init
                                         YARP::YARP_sig

